pkg/adapter/awscognitouserpoolsource/adapter_test.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright (c) 2019-2020 TriggerMesh Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package awscognitouserpoolsource
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider"
+	"github.com/aws/aws-sdk-go/service/cognitoidentityprovider/cognitoidentityprovideriface"
+
+	adaptertest "knative.dev/eventing/pkg/adapter/v2/test"
+	loggingtesting "knative.dev/pkg/logging/testing"
+)
+
+type mockedCognitoUserPoolClient struct {
+	cognitoidentityprovideriface.CognitoIdentityProviderAPI
+	listUsersOutput cognitoidentityprovider.ListUsersOutput
+}
+
+func (m mockedCognitoUserPoolClient) ListUsers(in *cognitoidentityprovider.ListUsersInput) (*cognitoidentityprovider.ListUsersOutput, error) {
+	return &m.listUsersOutput, nil
+}
+
+func TestListUsers(t *testing.T) {
+	user := cognitoidentityprovider.UserType{
+		UserLastModifiedDate: func() *time.Time {
+			now := time.Now()
+			return &now
+		}(),
```

Comment:
Using the `aws` [helper package](https://docs.aws.amazon.com/sdk-for-go/api/aws/#Time):
```suggestion
		UserLastModifiedDate: aws.Time(time.Now())
```